### PR TITLE
test(batch): add micro benchmark for NestedLoopJoinExecutor

### PR DESCRIPTION
**Pull request by wzzzzd**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Add micro benchmark for NestedLoopJoinExecutor and some minor changes.

Part of the benchmark results on ec2 c4.4xlarge instance:
```bash
NestedLoopJoinExecutor/32(Inner)                                                                            
                        time:   [449.19 ms 449.29 ms 449.39 ms]

NestedLoopJoinExecutor/128(Inner)                                                                            
                        time:   [355.96 ms 355.98 ms 356.00 ms]

NestedLoopJoinExecutor/512(Inner)                                                                            
                        time:   [333.32 ms 333.33 ms 333.35 ms]

NestedLoopJoinExecutor/1024(Inner)                                                                            
                        time:   [329.74 ms 329.76 ms 329.78 ms]

```

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/batch/benches/nested_loop_join.rs [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+async fn execute_nested_loop_join_executor(executor: BoxedExecutor) {
+    let mut stream = executor.execute();
+    while let Some(ret) = stream.next().await {
+        black_box(ret.unwrap());
+    }
+}
+
+fn bench_nested_loop_join(c: &mut Criterion) {
+    const LEFT_SIZE: usize = 1024;
```
Comment (liurenjie1024) [inline src/batch/benches/nested_loop_join.rs:165]: Too small?

File: src/batch/src/executor/test_utils.rs [hunk 2]
```diff
@@ -14,19 +14,57 @@
 
 use std::collections::VecDeque;
 use std::future::Future;
+use std::sync::Arc;
```
Comment (liurenjie1024) [inline src/batch/src/executor/test_utils.rs:17]: Move this file to `benches` folder, I think it's used by benches only.

**Comment by liurenjie1024** [comment]
Good job! Some small changes required.  BTW, could you share some data  of benchmark? You can run it on ec2.

**Comment by liurenjie1024** [comment]
Also, please sign CLA first.

File: src/batch/src/executor/test_utils.rs (hunk 2 above)
Comment (wzzzzd) [reply src/batch/src/executor/test_utils.rs:17]: It seems some unit tests of batch executors need utilities in this file.

Comment (wzzzzd) [reply src/batch/src/executor/test_utils.rs:17]: > Move this file to `benches` folder, I think it's used by benches only.

It seems some unit tests of batch executors need utilities in this file.

**Review by liurenjie1024 — APPROVED** [review]
LGTM

**Pull request merged**